### PR TITLE
[kcov] Bump version release 40

### LINKED ISSUE #11231: [request] kcov/40
### Package Details
  * Package Name/Version: **kcov/40**
  * Changelog: **https://github.com/SimonKagstrom/kcov/blob/master/ChangeLog**

Hello,

Currently conan-center provides only 38 release, I would like to have latest release (40) also available.
I'll provides a pull request.

### PR DESCRIPTION
**Pull request by EstebanDugueperoux2**
Specify library name and version:  **kcov/40**

closes #11231

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

**Comment by EstebanDugueperoux2** [comment]
Hi @uilianries,

I'm waiting access to https://ci-conan-prod.jfrog.team/job/prod/job/cci/job/PR-11233/4/display/redirect to fix failures.

Thanks in advance.

File: recipes/kcov/all/conanfile.py [hunk 1]
```diff
@@ -34,8 +35,9 @@ def source(self):
         os.rename(extracted_dir, self._source_subfolder)
 
     def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
+        if self.version == 38:
```
Comment (uilianries) [inline recipes/kcov/all/conanfile.py:38]: Revert. The conandata.yml is place to list which patch should be applied to which version.

**Comment by EstebanDugueperoux2** [comment]
@uilianries 
Now I get following error:

ERROR: [INCLUDE PATH DOES NOT EXIST (KB-H071)] Component kcov::kcov include dir 'include' is listed in the recipe, but not found in package folder. The include dir should probably be fixed or removed. (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H071)

See https://c3i.jfrog.io/c3i/misc/logs/pr/11233/9-linux-gcc/kcov/38//6a45c4b56e1770d80b4de07b3c126918157675a8-build.txt

The hook KB-H071 is not documented, i.e. https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H071 is a bad link.
In additition I have not include declaration in the recipe.

Could you help me to diagnostic this issue?

Regards.

**Comment by EstebanDugueperoux2** [comment]
I have just rebased but get randomly same error on kcov/38@ or kcov/40@.

**Comment by jgsogo** [comment]
The error from the hooks should be self-explanatory:

> [HOOK - conan-center.py] post_package_info(): ERROR: [INCLUDE PATH DOES NOT EXIST (KB-H071)] Component kcov::kcov include dir 'include' is listed in the recipe, but not found in package folder. The include dir should probably be fixed or removed. (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H071) 

If this package doesn´t provide anything in the `include` folder, you need to assign an empty list:

```python
def package_info(self):
   ...
   self.cpp_info.includedirs = []
```

**Comment by EstebanDugueperoux2** [comment]
> The error from the hooks should be self-explanatory:
> 
> > [HOOK - conan-center.py] post_package_info(): ERROR: [INCLUDE PATH DOES NOT EXIST (KB-H071)] Component kcov::kcov include dir 'include' is listed in the recipe, but not found in package folder. The include dir should probably be fixed or removed. (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H071)
> 
> If this package doesn´t provide anything in the `include` folder, you need to assign an empty list:
> 
> ```python
> def package_info(self):
>    ...
>    self.cpp_info.includedirs = []
> ```

Ok thanks, sorry but I'm beginner in all of that (conan, C++, ...).
I understand that by default self.cpp_info.includedirs as "include" default value.

Regards.

**Comment by uilianries** [comment]
Sorry, indeed we need to re-check hooks vs docs, because, from time to time, we update/fix and forget to update the documentation.

**Comment by EstebanDugueperoux2** [comment]
> Sorry, indeed we need to re-check hooks vs docs, because, from time to time, we update/fix and forget to update the documentation.

And what about "Error: open .c3i/c3i.yml: no such file or directory" error in https://github.com/conan-io/conan-center-index/runs/7100007371?check_suite_focus=true ?

**Comment by uilianries** [comment]
> And what about "Error: open .c3i/c3i.yml: no such file or directory" error in [conan-io/conan-center-index/runs/7100007371?check_suite_focus=true](https://github.com/conan-io/conan-center-index/runs/7100007371?check_suite_focus=true) ?

This was a bug, not a misinformation. It's fixed now.

**Comment by EstebanDugueperoux2** [comment]
\o/ all automatic checks are green.
Ready for manual approving.

**Review by uilianries — APPROVED** [review]

**Comment by EstebanDugueperoux2** [comment]
Thanks @uilianries for the approval.
Randomly I add another review to @madebr to unlock the merge.

**Review by danimtb — APPROVED** [review]

**Pull request merged**